File: backend/src/main/java/com/milktea/order/order/entity/Order.java
package com.milktea.order.order.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.math.BigDecimal;
import java.time.LocalDateTime;

/**
 * 订单主表（orders，LLD 2.3 数据模型 / V1__init_schema.sql）。
 *
 * <p>字段与建表脚本一一对应：六状态、三渠道来源、支付与时间组。
 * 订单项快照在 order_item 表（T10 落地下单时写入）；本实体在 T12 中被
 * 支付编排读取，并以「条件更新」推进 PENDING_PAYMENT → PAID。</p>
 */
@Data
@TableName("orders")
public class Order {

    @TableId(type = IdType.AUTO)
    private Long id;

    /** 订单号：yyMMdd + 5 位序列（T11 SequenceService 发放）。 */
    private String orderNo;

    /** 下单来源：MINI_PROGRAM / AI / COUNTER。 */
    private String source;

    /** 状态取值见 {@link OrderStatus}。 */
    private String status;

    /** 顾客 id；柜台单为 NULL。 */
    private Long customerId;

    /** 支付渠道：MOCK / WECHAT（预留），进入 PAID 时写入。 */
    private String payChannel;

    /** 渠道交易号，进入 PAID 时写入。 */
    private String transactionId;

    /** 应付总额（快照）。 */
    private BigDecimal totalAmount;

    /** 取餐码：进入 PAID 时分配（T11 SequenceService）。 */
    private String pickupCode;

    /** 口味备注。 */
    private String remark;

    /** 支付时间（进入 PAID）。 */
    private LocalDateTime paidAt;

    /** 开始制作时间。 */
    private LocalDateTime startedAt;

    /** 出餐时间。 */
    private LocalDateTime completedAt;

    /** 超时关闭时间。 */
    private LocalDateTime closedAt;

    /** 作废时间。 */
    private LocalDateTime voidedAt;

    /** 作废原因（商家填写）。 */
    private String voidReason;

    private LocalDateTime createdAt;

    private LocalDateTime updatedAt;
}

File: miniprogram/src/api/order.ts
import { request, type PageResult } from '@/utils/request'

/**
 * 顾客端订单接口（LLD 3.3 / 3.4）。
 * 下单与支付类接口需顾客 JWT（auth: true，request 层会自动补登录）。
 */

/** 订单项请求（只描述「要什么」，不含任何价格字段——价格一律后端计算） */
export type OrderItemRequest = {
  productId: number
  optionIds: number[]
  quantity: number
}

/** POST /api/customer/orders 请求体 */
export type OrderCreateRequest = {
  items: OrderItemRequest[]
  remark?: string
}

/** 下单响应中的订单项快照 */
export interface OrderCreatedItem {
  productName: string
  quantity: number
  unitPrice: string
  optionNames: string[]
  itemAmount: string
}

/** POST /api/customer/orders 响应 data */
export interface OrderCreateResult {
  id: number
  orderNo: string
  status: string
  /** 应付总额（两位小数字符串，后端计价引擎计算） */
  totalAmount: string
  /** 支付截止时间（创建时间 + 15 分钟） */
  expireAt: string
  items: OrderCreatedItem[]
}

/** POST /api/customer/orders/{id}/pay 响应 data */
export interface OrderPayResult {
  id: number
  status: string
  pickupCode: string
  paidAt: string
  payChannel: string
}

/** 轮询专用轻量状态（3 秒间隔） */
export interface OrderStatusResult {
  status: string
  pickupCode: string | null
  /** 该单之前处于 PAID/PREPARING 的单数（前端显示「排队第 seq+1 位」） */
  seq: number
}

/** 规格快照 */
export interface OptionSnapshot {
  groupName: string
  optionName: string
  priceDelta: string
}

/** 订单详情中的订单项 */
export interface OrderDetailItem {
  productId: number
  productName: string
  quantity: number
  basePrice: string
  unitPrice: string
  options: OptionSnapshot[]
  itemAmount: string
}

/** GET /api/customer/orders/{id} 响应 data */
export interface OrderDetail {
  id: number
  orderNo: string
  /** MINI_PROGRAM / AI / COUNTER */
  source: string
  status: string
  pickupCode: string | null
  totalAmount: string
  remark: string | null
  createdAt: string
  paidAt: string | null
  startedAt: string | null
  completedAt: string | null
  closedAt: string | null
  voidedAt: string | null
  voidReason: string | null
  items: OrderDetailItem[]
}

/** 订单列表项（active 与历史分页共用） */
export interface OrderListItem {
  id: number
  status: string
  pickupCode: string | null
  totalAmount: string
  createdAt: string
  /** 商品摘要，如 "珍珠奶茶x1(大杯/少冰/珍珠)" */
  items: string[]
  paidAt: string | null
  completedAt: string | null
  closedAt: string | null
  voidedAt: string | null
}

/** 创建订单（PENDING_PAYMENT），实付金额以返回的 totalAmount 为准 */
export function createOrder(data: OrderCreateRequest): Promise<OrderCreateResult> {
  return request<OrderCreateResult>({ url: '/api/customer/orders', method: 'POST', data, auth: true })
}

/** 发起支付（练手期为 Mock 通道，立即成功并分配取餐码） */
export function payOrder(orderId: number): Promise<OrderPayResult> {
  return request<OrderPayResult>({ url: `/api/customer/orders/${orderId}/pay`, method: 'POST', auth: true })
}

/** 进行中订单列表（PENDING_PAYMENT / PAID / PREPARING） */
export function getActiveOrders(): Promise<OrderListItem[]> {
  return request<OrderListItem[]>({ url: '/api/customer/orders/active', auth: true })
}

/** 历史订单分页（全部状态，创建时间倒序） */
export function getOrderHistory(page: number, size: number): Promise<PageResult<OrderListItem>> {
  return request<PageResult<OrderListItem>>({
    url: `/api/customer/orders?page=${page}&size=${size}`,
    auth: true
  })
}

/** 订单详情（含规格快照明细） */
export function getOrderDetail(orderId: number): Promise<OrderDetail> {
  return request<OrderDetail>({ url: `/api/customer/orders/${orderId}`, auth: true })
}

/** 轮询订单轻量状态 */
export function getOrderStatus(orderId: number): Promise<OrderStatusResult> {
  return request<OrderStatusResult>({ url: `/api/customer/orders/${orderId}/status`, auth: true })
}
